crates/burn-jit/src/bridge.rs

```diff
@@ -7,26 +9,34 @@ use core::marker::PhantomData;
 
 /// Handle precision conversion for the jit backend.
 #[derive(Debug)]
-pub struct PrecisionBridge<R> {
+pub struct PrecisionBridge<R, F: FloatElement, I: IntElement> {
     _runtime: PhantomData<R>,
+    _float_elem: PhantomData<F>,
+    _int_elem: PhantomData<I>,
 }
 
-impl<ROrigin, RTarget> BackendBridge<JitBackend<ROrigin>> for PrecisionBridge<RTarget>
+impl<ROrigin, FOrigin, IOrigin, RTarget, FTarget, ITarget>
+    BackendBridge<JitBackend<ROrigin, FOrigin, IOrigin>>
+    for PrecisionBridge<RTarget, FTarget, ITarget>
 where
     ROrigin: Runtime,
+    FOrigin: FloatElement,
+    IOrigin: IntElement,
     RTarget:
         Runtime<Device = ROrigin::Device, Server = ROrigin::Server, Channel = ROrigin::Channel>,
+    FTarget: FloatElement,
+    ITarget: IntElement,
 {
-    type Target = JitBackend<RTarget>;
+    type Target = JitBackend<RTarget, FTarget, ITarget>;
```

Comment:
It should be the same runtime, so in fact the `PrecisionBridge` should not be generic over the runtime trait.

Reply:
Whoops! That makes a lot of sense, the runtime generic is totally obsolete now that the element types are decoupled 😅